### PR TITLE
Add tc matchall filter

### PR DESCRIPTION
**Pull request by LKomaryanskiy**

### CODE REVIEW

File: arch/arm64/configs/defconfig [hunk 1]
```diff
@@ -161,7 +161,7 @@ CONFIG_IP6_NF_TARGET_MASQUERADE=m
 CONFIG_BRIDGE=m
 CONFIG_BRIDGE_VLAN_FILTERING=y
 CONFIG_NET_DSA=m
-CONFIG_VLAN_8021Q=m
+CONFIG_VLAN_8021Q=y
```
Comment (firscity) [inline arch/arm64/configs/defconfig]: Do we need this in this PR?

File: drivers/net/ethernet/renesas/rswitch_tc_filters.h [hunk 2]
```diff
@@ -25,8 +25,21 @@ struct rswitch_tc_filter {
 	enum rswitch_tc_action action;
 };
 
+static inline void rswitch_parse_pedit(struct rswitch_tc_filter *f, struct flow_action_entry *a)
+{
+	if (!a->mangle.offset) {
```
Comment (firscity) [inline drivers/net/ethernet/renesas/rswitch_tc_filters.h:34]: What this values mean?

File: drivers/net/ethernet/renesas/rswitch_tc_matchall.c [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+	/* Leave other paramters as zero */
+	cfg->param.priv = priv;
+	cfg->param.slv = 0x3F;
+
+	pf_param.rdev = rdev;
+	pf_param.all_sources = false;
+	pf_param.used_entries = 1;
+	pf_param.entries[0].val = 0;
+	pf_param.entries[0].mask = 0;
+	pf_param.entries[0].off = IPV4_DST_OFFSET;
```
Comment (firscity) [inline drivers/net/ethernet/renesas/rswitch_tc_matchall.c]: Why IPV4_DST_OFFSET?

File: drivers/net/ethernet/renesas/rswitch_tc_matchall.c [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+		cfg->param.l23_info.routing_number = rswitch_rn_get(priv);
+		cfg->param.l23_info.routing_port_valid = BIT(rdev->port) | BIT(filter->target_rdev->port);
+	}
+
+	pf_param.rdev = rdev;
+	pf_param.all_sources = false;
+	pf_param.used_entries = 1;
+	pf_param.entries[0].val = 0;
+	pf_param.entries[0].mask = 0;
+	pf_param.entries[0].off = IPV4_DST_OFFSET;
```
Comment (firscity) [inline drivers/net/ethernet/renesas/rswitch_tc_matchall.c]: Same question

File: drivers/net/ethernet/renesas/rswitch_tc_matchall.c [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+	struct rswitch_tc_filter filter = {
+		.action = 0,
+	};
+	int i;
+
+	filter.rdev = rdev;
+
+    flow_action_for_each(i, entry, actions) {
+		/* Several actions with drop cannot be offloaded */
+		if (filter.action != 0 && filter.action & ACTION_DROP)
```
Comment (firscity) [inline drivers/net/ethernet/renesas/rswitch_tc_matchall.c:119]: Tabulation

File: drivers/net/ethernet/renesas/rswitch_tc_matchall.c [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+	return 0;
+}
+
+static int rswitch_tc_matchall_replace(struct net_device *ndev,
+				  struct tc_cls_matchall_offload *cls_matchall)
+{
+    struct flow_action_entry *entry;
+    struct flow_action *actions = &cls_matchall->rule->action;
+    struct rswitch_device *rdev = netdev_priv(ndev);
+	struct rswitch_tc_filter filter = {
```
Comment (firscity) [inline drivers/net/ethernet/renesas/rswitch_tc_matchall.c:110]: Tabulation

File: drivers/net/ethernet/renesas/rswitch_tc_matchall.c [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+
+        if (entry->id == FLOW_ACTION_MANGLE &&
+            entry->mangle.htype == FLOW_ACT_MANGLE_HDR_TYPE_ETH) {
+            filter.action |= ACTION_SKBMOD;
+            rswitch_parse_pedit(&filter, entry);
+            continue;
+        }
+
+		/* Drop in hardware */
+		if (entry->id == FLOW_ACTION_DROP)
```
Comment (firscity) [inline drivers/net/ethernet/renesas/rswitch_tc_matchall.c:138]: Tabulation

File: drivers/net/ethernet/renesas/rswitch_tc_matchall.c [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+	case TC_CLSMATCHALL_REPLACE:
+		return rswitch_tc_matchall_replace(dev, cls_matchall);
+	case TC_CLSMATCHALL_DESTROY:
+		return rswitch_tc_matchall_destroy(dev, cls_matchall);
+	default:
+		return -EOPNOTSUPP;
+	}
+
+	return -EOPNOTSUPP;
+}
```
Comment (firscity) [inline drivers/net/ethernet/renesas/rswitch_tc_matchall.c]: Newline

File: drivers/net/ethernet/renesas/rswitch_tc_u32.c [hunk 9]
```diff
@@ -5,10 +5,6 @@
  * Copyright (C) 2022 EPAM Systems
  */
 
-#include <net/tc_act/tc_mirred.h>
```
Comment (firscity) [inline drivers/net/ethernet/renesas/rswitch_tc_u32.c]: Should it be in this PR?

File: arch/arm64/configs/defconfig (hunk 1 above)
Comment (LKomaryanskiy) [reply arch/arm64/configs/defconfig]: Removed config.

File: drivers/net/ethernet/renesas/rswitch_tc_filters.h (hunk 2 above)
Comment (LKomaryanskiy) [reply drivers/net/ethernet/renesas/rswitch_tc_filters.h:34]: Added comment with explanation

File: drivers/net/ethernet/renesas/rswitch_tc_matchall.c (hunk 3 above)
Comment (LKomaryanskiy) [reply drivers/net/ethernet/renesas/rswitch_tc_matchall.c]: Yes, it looks strange. As we need to match all packets, I've changed offset to zero value.

File: drivers/net/ethernet/renesas/rswitch_tc_matchall.c (hunk 4 above)
Comment (LKomaryanskiy) [reply drivers/net/ethernet/renesas/rswitch_tc_matchall.c]: The same

File: drivers/net/ethernet/renesas/rswitch_tc_matchall.c (hunk 5 above)
Comment (LKomaryanskiy) [reply drivers/net/ethernet/renesas/rswitch_tc_matchall.c:119]: Fixed

File: drivers/net/ethernet/renesas/rswitch_tc_matchall.c (hunk 6 above)
Comment (LKomaryanskiy) [reply drivers/net/ethernet/renesas/rswitch_tc_matchall.c:110]: Fixed

File: drivers/net/ethernet/renesas/rswitch_tc_matchall.c (hunk 7 above)
Comment (LKomaryanskiy) [reply drivers/net/ethernet/renesas/rswitch_tc_matchall.c:138]: Fixed

File: drivers/net/ethernet/renesas/rswitch_tc_matchall.c (hunk 8 above)
Comment (LKomaryanskiy) [reply drivers/net/ethernet/renesas/rswitch_tc_matchall.c]: Added

File: drivers/net/ethernet/renesas/rswitch_tc_u32.c (hunk 9 above)
Comment (LKomaryanskiy) [reply drivers/net/ethernet/renesas/rswitch_tc_u32.c]: The change has been removed from this PR. Thank you for the close review:)

**Comment by firscity** [comment]
LGTM

**Pull request merged**